src/olympia/devhub/tests/test_permissions.py

```diff
@@ -29,6 +31,13 @@ def test_has_permission_user_has_not_read_agreement(self):
             'The Firefox Add-on Distribution Agreement also links to our '
             'Privacy Notice which explains how we handle your information.')
 
+    def test_has_permission_disposable_email(self):
+        DisposableEmailDomainRestriction.objects.create(domain='example.com')
+        assert not self.permission.has_permission(self.request, self.view)
+        assert self.permission.message == (
+            'The email address you used for your developer account is not '
+            'allowed for add-on submission.')
```

Comment:
nit: `'The email address used for your account is not allowed for add-on submission.'`?

Reply:
Since we're already using that string and I think it was given by product I'm not changing it here. I do like your version better though, I'll file an issue about getting it changed (p5 :p)